src/main/java/mate/academy/AsyncRequestProcessor.java

```diff
@@ -1,16 +1,33 @@
 package mate.academy;
 
+import java.util.Map;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.Executor;
 
 public class AsyncRequestProcessor {
     private final Executor executor;
+    private final Map<String, UserData> cache = new ConcurrentHashMap<>();
 
     public AsyncRequestProcessor(Executor executor) {
         this.executor = executor;
     }
 
     public CompletableFuture<UserData> processRequest(String userId) {
-        return null;
+        if (cache.containsKey(userId)) {
+            return CompletableFuture.completedFuture(cache.get(userId));
+        }
+
+        return CompletableFuture.supplyAsync(() -> {
+            try {
+                Thread.sleep(1000); // Simulated delay
+            } catch (InterruptedException e) {
+                Thread.currentThread().interrupt();
+            }
```

Comment:
Simulating a long-running operation using `Thread.sleep` is not recommended in asynchronous code, as this could block the underlying thread from the executor's thread pool, reducing its ability to process other tasks. Consider using a scheduled executor service to simulate delay without blocking, or remove the sleep if it's just for demonstration purposes.